### ISSUE TITLE
Feature Request: getBTCAddress - Index Tracking System

### ISSUE THREAD

**Issue by EAWF** [EAWF/Bitcoin-Merchants-Toolbox#36]
### Hold for a future version:
- Add next_index system to automate the next index to be used in derivation. Might also need another field to indicate the last funded address.
- System would need to address the 20 address range issue in HD Wallets or addresses may be funded, but the wallet wouldn't "see" them.

**Comment by Septem151** [comment]
Fortunately we have all the tools capable to do this already with the toolbox functions. There could be a version of getBTCAddress that auto-increments to a most-recent unused address.

**Comment by EAWF** [comment]
You mean maintain 2 versions of getBTCAddress?  Is it maybe better to incorporate it now? There remains the matter of figuring out how to deal with the 20 address range of the wallets.  Maybe an additional field to be regularly updated that indicates the last funded address?  Still running it around in my mind.  Don't want to include too much program bloat that continuously delays rollout.

**Comment by Septem151** [comment]
> You mean maintain 2 versions of getBTCAddress? Is it maybe better to incorporate it now? 

No, I mean have a default parameter value set to null and in the function we can check if the value is null then use the Blockstream Esplora API to check, incrementally, the addresses starting from 0, or whatever predefined index, to find an address never used in a transaction before. Usage: `getBTCAddress($account);` as compared to choosing an index manually: `getBTCAddress($account, $index);`

> There remains the matter of figuring out how to deal with the 20 address range of the wallets. Maybe an additional field to be regularly updated that indicates the last funded address? Still running it around in my mind. Don't want to include too much program bloat that continuously delays rollout.

Why do we need to deal with the 20 address range of wallets? Not every wallet follows this scheme (ex: Wasabi) and the getBTCBalance function only looks up the balance for an address, not for an extended pubkey.

Additionally, there is no easy way to check a larger gap limit other than increasing it.

**Comment by EAWF** [comment]
OK, I understand not dealing with the address range...that's true, we don't care, and short of making this project a plugNplay paywall, I guess the onus is on the user to manage the indexes/addresses used, which then makes this issue moot. 
Maybe fork this repo off into a paywall project.

**Comment by Septem151** [comment]
In case we do decide to do the index autoincrementing, this was posted December 2019 on Esplora:
> we don't currently limit but we reserve the liberty to in the future, especially if the service is abused

But yeah, probably best to implement that only if you have a node yourself to run a private Esplora server on.

**Comment by EAWF** [comment]
Esplora's statement only addresses current, but I'm willing to bet a buck that it won't take long.
